bin/pre-commit.sh

```diff
@@ -13,6 +13,7 @@ fi
 js_files=$( git diff --diff-filter=d --staged --name-only | grep -E '^js\/\S*\.js$' )
 if [ ! -z "$js_files" ]; then
     npm run lint:js:files $js_files
+    npm run test:js $js_files -- --findRelatedTests
```

Comment:
In the pre-commit hook, this runs all Jest tests related to the staged files. So if only the `text.js` file is staged, it'll only run the test for that.